packages/celotool/src/lib/geth.ts

```diff
@@ -859,6 +859,10 @@ export async function startGeth(
   const privateKey = instance.privateKey || ''
   const lightserv = instance.lightserv || false
   const etherbase = instance.etherbase || ''
+  // Allow maxPeers to be zero
+  const maxPeers = instance.maxPeers !== undefined ? instance.maxPeers : gethConfig.instances.length
+  // Default to true
+  const setNodeKey = instance.setNodeKey !== undefined ? instance.setNodeKey : true
```

Comment:
Setting the node key by default is actually a relic that should have been removed already. We don't recommend it in the official docs, and we shouldn't do it here by default unless there is a reason I am not aware of.

Reply:
It's needed for proxies so that a proxied validator can specify the internal/external enode urls for its proxy when starting up without having to wait for the proxy to be deployed first.

I just made setNodeKey default to being falsy, and force the node key to be set when the node is a proxy. We can decide later on if it's worth changing up the e2e tests to remove the need entirely